### PR TITLE
feat: Add firmware update progress reporting to update platform

### PR DESCRIPTION
**Pull request by firstof9**
## Description

This PR adds progress reporting features to the `update` platform using the `python-openevse-http` 0.4.1 library.
Specifically, it adds:
- `UpdateEntityFeature.PROGRESS` support to `OpenEVSEUpdateEntity`.
- `in_progress` and `update_percentage` properties to retrieve firmware update status.
- A background task `_async_poll_update_progress` that polls `/status` to track update progress since progress updates are not pushed over WebSockets.
- Fully formatted and passing tests checking attributes and update polling loop progression.

Fixes # (issue)

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

## Summary by CodeRabbit

* **New Features**
  * Background progress polling added for firmware installations to provide continuous update status and improved visibility during installs.

* **Bug Fixes**
  * Polling now tolerates transient errors and device reboot/connectivity interruptions without disrupting the install flow.

* **Tests**
  * Added tests covering polling behavior, error handling, and final install state.

* **Chores**
  * Bumped bundled library dependency for OpenEVSE HTTP client.

### CODE REVIEW

**Comment by firstof9** [comment]
@coderabbitai review

**Comment by secondof9** [comment]
## Code Review — PR #631: Add firmware update progress reporting

### Overview
This PR adds background progress polling for firmware updates using the `python-openevse-http` 0.4.2 library, along with `UpdateEntityFeature.PROGRESS` support and `in_progress`/`update_percentage` attributes. All tests pass.

---

### 🟢 File: `custom_components/openevse/manifest.json`

| Line | Change | Review |
|------|--------|--------|
| 19 | Bumped `python-openevse-http` from `0.4.1` → `0.4.2` | ✅ The dependency bump aligns with the PR's use of the new library version. Verify the release notes confirm no breaking API changes. |

---

### 🟡 File: `custom_components/openevse/update.py`

| Line | Change | Review |
|------|--------|--------|
| 6 | Added `from asyncio import sleep as async_sleep` | ✅ Standard HA pattern for non-blocking async sleep. Consider adding a brief inline comment explaining the alias purpose. |
| 142–146 | Added `self.hass.async_create_background_task(self._async_poll_update_progress(), "openevse_update_progress_polling",)` | ✅ Correctly spawns a background task with a name for debugging/logging. |
| 148–163 | Implemented `_async_poll_update_progress()` | ✅ Background polling loop polls `/status` every 2 seconds, tolerates transient errors, and stops when `ota_update=0`. Properly calls `coordinator.websocket_update()` after each poll to push state changes. |

**Code Review Findings:**
- ✅ The polling loop uses a bounded `for _ in range(150)` guard, preventing infinite loops if the device remains in update mode unexpectedly.
- ✅ Errors during polling are logged at `DEBUG` level, which is appropriate since transient failures are expected during reboots or connectivity blips.
- ✅ The polling stops when `not self._manager.ota_update`, correctly terminating the loop once the update completes.

---

### 🟢 File: `requirements.txt`

| Line | Change | Review |
|------|--------|--------|
| 1 | Bumped `python-openevse-http==0.4.1` → `0.4.2` | ✅ Matches the manifest.json bump. |

---

### 🟢 File: `tests/test_update.py`

| Line | Change | Review |
|------|--------|--------|
| 101–106 | Updated `test_update_install()` to patch `async_sleep` and call `async_block_till_done()` | ✅ Patching `async_sleep` is necessary for test determinism. Calling `async_block_till_done()` ensures the background task completes before assertions. |
| 198–268 | Added `test_update_polling()` | ✅ New test covers the polling loop behavior, including error handling and state transitions. The mock properly tests a successful poll, a transient error (to test recovery/reboot), and the final completion poll. |

**Code Review Findings:**
- ✅ The test uses `async_block_till_done()` to wait for the background task to complete, ensuring deterministic test behavior.
- ✅ The test correctly patches `async_sleep` to avoid real time delays during test execution.
- ✅ The assertions verify that `poll_count == 3` after the test, confirming the polling loop ran exactly as expected.

---

### ✅ Overall Assessment
This PR is ready to merge. The code is clean, well-tested, and properly handles edge cases like device reboots and connectivity interruptions. The background polling loop is bounded and gracefully stops when the update completes. All tests pass, and the dependency bump is correctly reflected in both manifest.json and requirements.txt.

**Suggestions:**
1. Add a brief inline comment in update.py:6 explaining the purpose of the `async_sleep` alias.
2. Verify the release notes for `python-openevse-http` 0.4.2 to ensure no unexpected API changes.

---

**Verdict:** 🟢 LGTM — No critical issues, only minor suggestions for documentation.

File: requirements.txt [hunk 1]
```diff
@@ -1 +1 @@
-python-openevse-http==0.4.1
+python-openevse-http==0.4.2
```
[the remark this answers is not in the document]
Comment (firstof9) [reply requirements.txt:1]: This is an automated scanner mismatch report. In Home Assistant custom integrations, we do not lock or pin transitive dependencies like `aiohttp` in `requirements.txt` (as `aiohttp` is provided directly by the Home Assistant core environment at runtime). Pinning transitive dependencies would conflict with the Home Assistant core package requirements.
